### PR TITLE
Include dict

### PR DESCRIPTION
**Pull request by daniel-sol**
Suggested solution to issue #354

### CODE REVIEW

File: src/fmu/dataio/_definitions.py [hunk 1]
```diff
@@ -32,6 +33,7 @@ def __post_init__(self):
             "csv|xtgeo": ".csv",  # use default xtgeo columns: X_UTME, Y_UTMN, Z_TVDSS
             "irap_ascii": ".poi",
         }
+        self.dictionary = {"json": ".json", "yaml": ".yml"}
```
Comment (perolavsvendsen) [inline src/fmu/dataio/_definitions.py:36]: Good example of something that will be potentially very difficult to understand 6 months from now, without the context. If we move on with the `dictionary` as a class, we probably need to explain this with comments. If we have no better options, we will have to go with it, but this is a major code smell I think.

**Review by perolavsvendsen — CHANGES_REQUESTED** [review]
Need tests that clearly show how we intend this to be used. At least some unit tests. Suggestions:
- Have dict instance, export it with metadata. Assert results are as expected.
- Open JSON file from disk, create metadata
- Open non-JSON file from disk, turn it into JSON, create metadata
- Try to pass a dictionary that cannot be turned into JSON - how does the code handle that, and what error messages are given?

Specific use case is `parameters.txt` I believe. Perhaps add that explicitly as a test, which will be something between unit testing and integration testing I guess.

File: src/fmu/dataio/_objectdata_provider.py [hunk 2]
```diff
@@ -486,6 +497,15 @@ def _derive_spec_bbox_arrowtable(self):
 
         return specs, bbox
 
+    def _derive_spec_bbox_dict(self):
+        """Process/collect the data items for Arrow table."""
+        logger.info("Process data metadata for arrow (tables)")
```
Comment (equinor-ruaj) [inline src/fmu/dataio/_objectdata_provider.py:502]: Is this docstring and logging correct?

Comment (daniel-sol) [reply src/fmu/dataio/_objectdata_provider.py:502]: Nope, well spotted. Changed now..

**Review by equinor-ruaj — APPROVED** [review]
lgtm

**Review by perolavsvendsen — APPROVED** [review]
OK let's try it and see how it works. Gut feeling that this might be confusing, but I don't see any better options. This PR should probably include a documentation update, but I guess there are more that should be included there... 

This PR should trigger a new version of fmu-dataio.

**Pull request merged**